tests/phpt/error_handler_respects_current_error_reporting_level.phpt

```diff
@@ -0,0 +1,79 @@
+--TEST--
+Test that the error handler uses the current error_reporting() level.
+--INI--
+error_reporting=E_ALL
+display_errors=Off
+--FILE--
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Tests;
+
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Promise\PromiseInterface;
+use Sentry\ClientBuilder;
+use Sentry\Event;
+use Sentry\Options;
+use Sentry\Response;
+use Sentry\ResponseStatus;
+use Sentry\SentrySdk;
+use Sentry\Transport\TransportFactoryInterface;
+use Sentry\Transport\TransportInterface;
+
+$vendor = __DIR__;
+
+while (!file_exists($vendor . '/vendor')) {
+    $vendor = \dirname($vendor);
+}
+
+require $vendor . '/vendor/autoload.php';
+
+$transportFactory = new class implements TransportFactoryInterface {
```

Comment:
Wouldn't a `before_send` callback obtain the same objective and be less verbose?

Reply:
Yeah sure. I just copied this from a few other tests, but I'll change it 👍